From cb5924b5a7d7fe47c32d74240800062add7adf66 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Mon, 22 Apr 2024 06:46:36 +0200
Subject: [PATCH] bytesatwork: imx8mp: introduce bawconfig

Configure defaults for bytes at work tool to store production data
in EEPROM.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
---
 board/bytesatwork/bytedevkit_imx8mp/Kconfig         |  2 ++
 board/bytesatwork/bytedevkit_imx8mp/Makefile        |  2 ++
 .../bytedevkit_imx8mp/imx8mp_bytedevkit.c           | 13 +++++++++++++
 board/bytesatwork/bytedevkit_imx8mp/spl.c           |  5 +++++
 include/configs/imx8mp_bytedevkit.h                 |  4 ++++
 5 files changed, 26 insertions(+)

diff --git a/board/bytesatwork/bytedevkit_imx8mp/Kconfig b/board/bytesatwork/bytedevkit_imx8mp/Kconfig
index cff47a7922..d53eb6c991 100644
--- a/board/bytesatwork/bytedevkit_imx8mp/Kconfig
+++ b/board/bytesatwork/bytedevkit_imx8mp/Kconfig
@@ -12,4 +12,6 @@ config SYS_CONFIG_NAME
 config IMX_CONFIG
 	default "board/bytesatwork/bytedevkit_imx8mp/imximage-8mp-lpddr4.cfg"
 
+source "board/bytesatwork/baw_config/Kconfig"
+
 endif
diff --git a/board/bytesatwork/bytedevkit_imx8mp/Makefile b/board/bytesatwork/bytedevkit_imx8mp/Makefile
index a2505271d1..7d3a4eaf38 100644
--- a/board/bytesatwork/bytedevkit_imx8mp/Makefile
+++ b/board/bytesatwork/bytedevkit_imx8mp/Makefile
@@ -18,3 +18,5 @@ obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing.o
 obj-$(CONFIG_IMX8M_DDR4) += ddr4_timing.o
 endif
 endif
+
+obj-y += ../baw_config/
diff --git a/board/bytesatwork/bytedevkit_imx8mp/imx8mp_bytedevkit.c b/board/bytesatwork/bytedevkit_imx8mp/imx8mp_bytedevkit.c
index 8563255547..fcc41b83e6 100644
--- a/board/bytesatwork/bytedevkit_imx8mp/imx8mp_bytedevkit.c
+++ b/board/bytesatwork/bytedevkit_imx8mp/imx8mp_bytedevkit.c
@@ -31,11 +31,13 @@
 #include <imx_sip.h>
 #include <linux/arm-smccc.h>
 #include <mmc.h>
+#include "../baw_config/baw_config_eeprom.h"
 
 DECLARE_GLOBAL_DATA_PTR;
 
 #define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1)
 #define WDOG_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_ODE | PAD_CTL_PUE | PAD_CTL_PE)
+#define I2C_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_HYS | PAD_CTL_PUE | PAD_CTL_PE)
 
 static iomux_v3_cfg_t const uart_pads[] = {
 	MX8MP_PAD_UART1_RXD__UART1_DCE_RX | MUX_PAD_CTRL(UART_PAD_CTRL),
@@ -46,6 +48,11 @@ static iomux_v3_cfg_t const wdog_pads[] = {
 	MX8MP_PAD_GPIO1_IO02__WDOG1_WDOG_B  | MUX_PAD_CTRL(WDOG_PAD_CTRL),
 };
 
+static const iomux_v3_cfg_t i2c1_pads[] = {
+	MX8MP_PAD_I2C1_SCL__I2C1_SCL | MUX_PAD_CTRL(I2C_PAD_CTRL),
+	MX8MP_PAD_I2C1_SDA__I2C1_SDA | MUX_PAD_CTRL(I2C_PAD_CTRL),
+};
+
 #ifdef CONFIG_NAND_MXS
 
 static void setup_gpmi_nand(void)
@@ -83,6 +90,8 @@ int board_early_init_f(void)
 
 	init_uart_clk(0);
 
+	imx_iomux_v3_setup_multiple_pads(i2c1_pads, ARRAY_SIZE(i2c1_pads));
+
 	return 0;
 }
 
@@ -497,6 +506,10 @@ int board_late_init(void)
 	env_set("board_rev", "iMX8MP");
 #endif
 
+	if (IS_ENABLED(CONFIG_BAW_CONFIG_EEPROM))
+		if (baw_config_eeprom_init())
+			printf("EEPROM init failed!\n");
+
 	return 0;
 }
 
diff --git a/board/bytesatwork/bytedevkit_imx8mp/spl.c b/board/bytesatwork/bytedevkit_imx8mp/spl.c
index 750631c3f8..a05441cfe6 100644
--- a/board/bytesatwork/bytedevkit_imx8mp/spl.c
+++ b/board/bytesatwork/bytedevkit_imx8mp/spl.c
@@ -29,6 +29,7 @@
 #include <fsl_esdhc_imx.h>
 #include <mmc.h>
 #include <asm/arch/ddr.h>
+#include "../baw_config/baw_config_eeprom.h"
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -60,6 +61,10 @@ int spl_board_boot_device(enum boot_device boot_dev_spl)
 
 void spl_dram_init(void)
 {
+	if (IS_ENABLED(CONFIG_BAW_CONFIG_EEPROM))
+		if (baw_config_eeprom_init())
+			printf("EEPROM init failed!\n");
+
 	ddr_init(&dram_timing);
 }
 
diff --git a/include/configs/imx8mp_bytedevkit.h b/include/configs/imx8mp_bytedevkit.h
index 12a5cd91a9..ed876e71c9 100644
--- a/include/configs/imx8mp_bytedevkit.h
+++ b/include/configs/imx8mp_bytedevkit.h
@@ -15,6 +15,10 @@
 #include <asm/arch/imx-regs.h>
 #include "imx_env.h"
 
+#define BAW_CONFIG_BUILTIN_PCB		M7_PCB_REV_1_0
+#define BAW_CONFIG_BUILTIN_RAM		M7_RAM_MT53E1G32D2FW_046
+#define BAW_CONFIG_BUILTIN_FLASH	EMMC_16GB
+
 #define CONFIG_SYS_BOOTM_LEN		(32 * SZ_1M)
 
 #define CONFIG_SPL_MAX_SIZE		(176 * 1024)
-- 
2.39.2

